src/annotator/frame-observer.js

```diff
@@ -126,11 +126,7 @@ export function onDocumentReady(frame) {
       frame.hasAttribute('src') &&
       frame.src !== 'about:blank'
     ) {
-      // Unfortunately, listening for 'DOMContentLoaded' on the iframeDocument
-      // doesn't work. Instead, we need to wait for a 'load' event to be triggered.
-      frame.addEventListener('load', () => {
-        resolve();
-      });
+      setTimeout(() => onDocumentReady(frame).then(resolve).catch(reject), 10);
```

Comment:
Are there situations where this could loop indefinitely? If so it might be wise to add a timeout to the whole process. Also, how long does the initial navigation typically take? I'm wondering how many recursive calls to `onDocumentReady` will happen in the typical case.

Reply:
Actually, I was able to create a situation where the loop runs indefinitely. In Safari (and sometimes in Chrome), if the iframe has an unreachable site in the `src` (like 'http://unreachable.gov'), the iframe stays in the initial `about:blank` document, causing an infinite loop. 

Here is a video:

https://user-images.githubusercontent.com/8555781/133477444-daf92eff-eba7-4d49-9bd9-085cf5bcc99a.mov

I include an additional commit that includes a fix for that.

Reply:
For a Chrome and Firefox any document (of any size) takes 1 or 2 calls.

For Safari, it seems that it is related to the size of the document. For medium documents takes 1 or 2 recursive calls. For large documents (https://html.spec.whatwg.org/), it can take up to 40 calls with my internet speed. What would happened when offline?